src/App.tsx

```diff
@@ -1,19 +1,152 @@
 import React from 'react';
-import './App.css';
+import './App.scss';
 
 import users from './api/users';
+import todos from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+const preparedTodos = todos.map((todo) => {
+  return {
+    ...todo,
+    user: users.find(user => user.id === todo.userId) || null,
+  };
+});
+
+type State = {
+  todosList: Todo[],
+  taskText: string,
+  selectedUser: string,
+  userNotSelectedMessage: boolean,
+  titleIsEmpty: boolean,
 
-const App: React.FC = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    todosList: [...preparedTodos],
+    taskText: '',
+    selectedUser: '',
+    userNotSelectedMessage: false,
+    titleIsEmpty: false,
+  };
+
+  clearForm = () => {
+    this.setState({ taskText: '', selectedUser: '' });
+  };
+
+  validateForm = () => {
+    if (this.state.selectedUser !== '') {
+      this.setState({ userNotSelectedMessage: false });
+    } else {
+      this.setState({ userNotSelectedMessage: true });
+    }
```

Comment:
How about this?
```suggestion
     this.setState({ userNotSelectedMessage: this.state.selectedUser === ''"})
```

Also, I think it'd be better if the name was hasUserSelectedMessage, without the negative allocation;)

Reply:
you are so damn right about that!) changed it)